### ISSUE TITLE
\code blocks have blank lines removed in LaTeX/PDF output

### ISSUE THREAD

**Issue by dhowland** [doxygen/doxygen#9721]
**Describe the bug**
When my documentation uses` \code{.c}` `\endcode` blocks to represent source code, the PDF output strips out the blank lines, therefore degrading the readability of the code.

**Expected behavior**
I expect the code block to be represented exactly as preformatted monspace text.

**Screenshots**
So this:
```
int myfunc(void) {
  int x=0;

  return x;
}
```
Is printed like this in the resulting PDF:
```
int myfunc(void) {
  int x=0;
  return x;
}
```

**To Reproduce**
I cannot post my proprietary document.

**Version**
Doxygen 1.9.5
Texlive 20220321
Cygwin 

**Stack trace**
N/A

**Additional context**
The `.tex` output does contain the `\DoxyCodeLine{}` commands in the correct places, however they do not result in empty lines being printed in the resulting document.  Adding spaces to the blank lines will result in `\DoxyCodeLine{ }` but still no luck.  I'm not familiar with the LaTeX language, so I don't know what commands are used to print a blank line in this context.

**Comment by albert-github** [comment]
Indeed between the HTML version and the LaTeX / PDF version there is a small difference.
When having:
```
/** \file


\code{.c}
int myfunc(void) {
  int x=0;

  return x;
}
\endcode
*/
```
and a default `Doxyfile`.

We see in the HTML version:
![image](https://user-images.githubusercontent.com/5223533/207561789-2f6b0822-49cf-4839-a27f-ae18bf848d14.png)

and in the LaTeX / HTML version:
![image](https://user-images.githubusercontent.com/5223533/207562111-58dd1784-cd43-460c-961c-abbc1e45845e.png)

Most likely a `\newline` has to be placed in `\DoxyCodeLine` between the brackets. Have to think whether this should be done in the code (latexdocvisitor.cpp / latexgen.cpp or in the defiunition of the `\DoxyCodeLine` as an `if` / `else` construct.


(Note The description of the problem was in such a way I could reconstruct the problem, but in future please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.)

**Comment by dhowland** [comment]
Thanks for looking at it, it is very much appreciated.

I manually edited the `.tex` to convert the lines to `\DoxyCodeLine{\newline}` and it does work, however it's more like two lines of space being inserted which looks strange in a different way.  Using `\\[1em]` causes an error (I don't know what I'm doing)

**Comment by dhowland** [comment]
Ok last update I replaced `\DoxyCodeLine{}` with `\DoxyCodeLine{\phantom{;}}` and it looks perfect.

**Comment by dhowland** [comment]
More information:  I'm using tables to put code blocks side-by-side and I think that being inside the table there is some interaction with the code blocks that might be causing this because in that arrangement there is also a problem with indents.  Only the first level of code indentation works and everything else is flattened.

**Comment by albert-github** [comment]
@dhowland regarding the comments in https://github.com/doxygen/doxygen/issues/9721#issuecomment-1351695997 (interaction with tables), please create a small example otherwise this it is hard to comment. Actually you might need 2 examples, for the problems you mention there.

The idea of the `\phantom` is very interesting, though as can be seen in https://tex.stackexchange.com/a/212712/44119 it looks like to create some space (which might not be a problem though).
After some research I think I have a better solution using `vspace` and `\baselineskip`. This has to be integrated into the `doxygen.sty` (first tests look quite OK, but some more tests are required), integrating in `latexgen.cpp` would be very difficult.

The initial idea of the `\newline` indeed gives a bit a wide blank gap.

**Comment by dhowland** [comment]
Here is the example code:

```
/**
\page sec7_test Testing

<table>
<tr><th>Incorrect</th>
<th>Correct</th></tr>
<tr><td>
\code{.c}
int myFunc(int x, int y) {
    int z;
    
    if (x == 0) {
        z = y;
    }
    
    return z;
}
\endcode
</td><td>
\code{.c}
int myFunc(int x, int y) {
    int z = 0;
    
    if (x == 0) {
        z = y;
    }
    
    return z;
}
\endcode
</td></tr>
</table>
*/
```

Here is the HTML output, looks good:
![test_html](https://user-images.githubusercontent.com/1534621/207700457-1909e021-87ba-48ed-bc84-8b4ff82a392d.png)

Here is the PDF output, with both missing blank lines and missing indents

![test_pdf](https://user-images.githubusercontent.com/1534621/207700467-381ae61d-b624-46c5-beba-047f9d9ddfbe.png)

I have actually solved both problems by making a tool to pre-process the `.tex` files and replace leading whitespace in DoxyCodeLines with equivalent length \phantom{} commands

**Comment by dhowland** [comment]
For completeness, here is my hacked up latex:
```
\begin{DoxyCode}{0}
\DoxyCodeLine{\textcolor{keywordtype}{int} myFunc(\textcolor{keywordtype}{int} x, \textcolor{keywordtype}{int} y) \{}
\DoxyCodeLine{\phantom{....}\textcolor{keywordtype}{int} z = 0;}
\DoxyCodeLine{\phantom{....}}
\DoxyCodeLine{\phantom{....}\textcolor{keywordflow}{if} (x == 0) \{}
\DoxyCodeLine{\phantom{........}z = y;}
\DoxyCodeLine{\phantom{....}\}}
\DoxyCodeLine{\phantom{....}}
\DoxyCodeLine{\phantom{....}\textcolor{keywordflow}{return} z;}
\DoxyCodeLine{\}}
```

and here is my 'fixed' PDF output:
![image](https://user-images.githubusercontent.com/1534621/207701465-f5d54e5b-c952-4571-b992-c57a0720ad8f.png)

**Comment by albert-github** [comment]
The examples did show some extra problems (though the words `correct` and `incorrect` were a it confusing as they were not related to doxygen but to the code style to be used in the package). There was also a, small, indentation problem regarding when showing showing line numbers at some places, so the showing of the line numbers has been made uniform as well.

I've just pushed a proposed patch, pull request #9724

**Comment by dhowland** [comment]
Sorry for the confusion.  My intention was to show why I would want to have code blocks side-by-side in a table.  (In this case, it's a simple demonstration of an uninitialized variable.)

Your work is appreciated!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
